dGame/dComponents/PetComponent.cpp

```diff
@@ -158,7 +157,7 @@ void PetComponent::OnUse(Entity* originator) {
 		return;
 	}
 
-	if (m_Preconditions != nullptr && !m_Preconditions->Check(originator, true)) {
+	if (m_Preconditions.has_value() && !m_Preconditions->Check(originator, true)) {
```

Comment:
```suggestion
	if (m_Preconditions && !m_Preconditions->Check(originator, true)) {
```
optional should be implicitly convertible to bool so the has_value can be dropped

Reply:
The codebase standard has been pretty consistent with utilizing operator bools on optional.  Not sure why specifically here you are drawing the line with readability.